File: app/src/main/java/com/chooloo/www/callmanager/ui/activity/contact/ContactContract.java
package com.chooloo.www.callmanager.ui.activity.contact;

import com.chooloo.www.callmanager.ui.base.BaseContract;

public class ContactContract {
    public interface View extends BaseContract.View {

        void setUp();

        void actionCall();

        void actionSms();

        void actionEdit();

        void actionInfo();

        void actionDelete();

        void actionFav();

        void toggleFavIcon(boolean toggle);

        void setContactFromIntent();

        void handleNoRecents();

        void onBackPressed();
    }

    public interface Presenter<V extends View> extends BaseContract.Presenter<V> {

        void onBackButtonPressed();

        void onActionCall();

        void onActionSms();

        void onActionEdit();

        void onActionInfo();

        void onActionDelete();

        void onActionFav();
    }
}

File: app/src/main/java/com/chooloo/www/koler/ui/contact/ContactContract.kt
package com.chooloo.www.koler.ui.contact

import android.net.Uri
import com.chooloo.www.koler.ui.base.BaseContract

interface ContactContract : BaseContract {
    interface View : BaseContract.View {
        val contactId: Long
        var contactName: String?
        var contactImage: Uri?
        var isStarIconVisible: Boolean
    }

    interface Controller<V : View> : BaseContract.Controller<V> {
        fun onActionSms()
        fun onActionCall()
        fun onActionEdit()
        fun onActionInfo()
        fun onActionDelete()
    }
}